lib/ansible/plugins/action/__init__.py

```diff
@@ -549,7 +563,7 @@ def _execute_module(self, module_name=None, module_args=None, tmp=None, task_var
 
         # Fix permissions of the tmp path and tmp files.  This should be
         # called after all files have been transferred.
-        self._fixup_perms(tmp, remote_user, recursive=True)
+        self._fixup_perms(tmp, remote_user, recursive=True, execute=is_binary)
```

Comment:
I've been thinking perhaps we should always set execute=True?  In the python module case, it is still a script intended to be executed so it seems reasonable.